### PR TITLE
Core initial implementation

### PR DESCRIPTION
**Pull request by eerofeev**

### CODE REVIEW

File: cheetunit-core/src/main/java/com/gepardec/cheetunit/core/CheetUnitExecuter.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import javax.enterprise.inject.spi.BeanManager;
+import javax.inject.Inject;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.io.Serializable;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.*;
+
+public class CheetUnitExecuter {
```
Comment (patholzer) [inline cheetunit-core/src/main/java/com/gepardec/cheetunit/core/CheetUnitExecuter.java:21]: better rename to CheetUnitExecutor

File: cheetunit-core/src/main/java/com/gepardec/cheetunit/core/CheetUnitClassloader.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    public void feedClasses(Map<String, byte[]> classes) {
+        if (knownClasses == null) {
+            knownClasses = new HashMap<>();
+        }
+
+        knownClasses.putAll(classes);
+    }
+
+    @Override
+    protected Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
```
Comment (patholzer) [inline cheetunit-core/src/main/java/com/gepardec/cheetunit/core/CheetUnitClassloader.java:29]: JavaDoc would be appreciated here, as we already had a clearifing discussion for this method

File: cheetunit-test/src/main/java/com/gepardec/cheetunit/test/RequestDTOFactory.java [hunk 3]
```diff
@@ -5,24 +5,24 @@
...
 
 import java.util.Base64;
 import java.util.List;
 import java.util.Map;
 
 /**
- * Factory for creating a new object of type {@link RequestDTO}
+ * Factory for creating a new object of type {@link ExecutionRequest}
  */
 class RequestDTOFactory {
```
Comment (patholzer) [inline cheetunit-test/src/main/java/com/gepardec/cheetunit/test/RequestDTOFactory.java:18]: Rename to ExecutionRequestFactory

File: cheetunit-test/pom.xml [hunk 4]
```diff
@@ -15,12 +15,7 @@
     <dependencies>
         <dependency>
             <groupId>com.gepardec.cheetunit</groupId>
-            <artifactId>cheetunit-core</artifactId>
-            <version>${project.version}</version>
-        </dependency>
-        <dependency>
-            <groupId>com.gepardec.cheetunit</groupId>
-            <artifactId>cheetunit-api</artifactId>
+            <artifactId>cheetunit-wildfly</artifactId>
```
Comment (patholzer) [inline cheetunit-test/pom.xml:18]: use cheetunit-core instead of cheetunit-wildfly

**Review by patholzer — CHANGES_REQUESTED** [review]

**Review by csalmhof — APPROVED** [review]

**Review by patholzer — APPROVED** [review]

**Pull request merged**